common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -114,14 +114,29 @@
         (es-helper/search
          (context->conn context) (:index-name index-info) [(:type-name index-info)] query))
       (errors/throw-service-error :service-unavailable "Exhausted retries to execute ES query"))
+
     (catch UnknownHostException e
       (info (format
              (str "Execute ES query failed due to UnknownHostException. Retry in %.3f seconds."
                   " Will retry up to %d times.")
              (/ (es-config/elastic-unknown-host-retry-interval-ms) 1000.0)
              max-retries))
       (Thread/sleep (es-config/elastic-unknown-host-retry-interval-ms))
-      (do-send-with-retry context index-info query (- max-retries 1)))))
+      (do-send-with-retry context index-info query (- max-retries 1)))
+
+    (catch clojure.lang.ExceptionInfo e
+      (when-let [body (:body (ex-data e))]
+        (when (re-find #"Trying to create too many scroll contexts" body)
+          (info "Execute ES query failed due to" body)
+          (errors/throw-service-error
+           :too-many-requests
+           "Trying to create too many scroll contexts.
```

Comment:
I would change this line to something like "CMR is currently experiencing too many scroll searches to complete this request."